contributors/design-proposals/priority-eviction.md

```diff
@@ -0,0 +1,62 @@
+
+The current method for ranking pods for eviction is by QoS, then usage over requests.  This currently holds the invariant that a pod that does not exceed its requests is not evicted, since the sum of the requests on the node cannot exceed the allocatable memory on the node.
+
+If the kubelet is unable to respond to memory pressure in time, an OOM Kill may be triggered.  In this case, processes are killed based on their OOM Score.  See the [OOM Score configuration docs](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-qos.md#oom-score-configuration-at-the-nodes) for more details.
+
+### Goals
+ - Transparency.  Users should be able to understand why their pods are evicted, and what, if anything, they can do to prevent future evictions.
+ - Low Abuse.  High priority pods should not be able to intentionally, or unintentionally disrupt large numbers of well-behaved pods.
+ - Respect Priority.  Pods that have higher priority should be less likely to be evicted.
+
+The goal of this proposal is to build consensus on the general design of how priority is integrated with memory eviction.  This proposal itself will not be merged, but will result in a set of changes to the [kubelet eviction documentation](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/kubelet-eviction.md) after a broad design is settled on.
+
+### Non Goals
+The implementation of priority itself is outside the scope of this proposal, and is covered in the [Priority](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/pod-priority-api.md) proposal.  This includes mechanisms to control which pods are allowed to be given which priority levels.
+
+The scope of this design is restricted to the node, and does not make any proposals regarding cluster-level controllers.
+
+## Proposed Implementation
+
+### Only evict pods where usage > requests.  Then sort by Function(priority, usage - requests)
```

Comment:
I thought we settled down with a simpler algorithms in the meeting:

Only evict pods where usage > requests. Among those pods in eviction pool, we first sort them by priority, then sort by ( usage - request ) as the tie-breaker for the pods at the same priority. 

I liked the initial proposal way better than introducing a function to re-generate an internal priority for eviction because:

1) It is much more deterministic. If there is a resource starvation issue, the user expect the abusive pods (with usage > request) with lower-priority will be evicted before the higher priority ones. 
2) It might introduce reverse-priority issue to the system. 
3) It is hard to explain to the users upon a pod being evicted at a given time. 
4) The same priority is used for preemption by the scheduler(s) / rescheduler(s) / controller(s). The current proposal might introduce a contradict decision for the eviction on the node. 

cc/ @dashpole @derekwaynecarr @bsalamat @kubernetes/sig-node-proposals

Reply:
cc/ @vishh

Reply:
LGTM